pkg/apis/cluster/v1alpha2/machine_types.go

```diff
@@ -163,11 +163,38 @@ type MachineStatus struct {
 	// +optional
 	Phase *string `json:"phase,omitempty"`
 
-	// Bootstrap is the state of the bootstrap provider.
-	Bootstrap *string `json:"bootstrap,omitempty"`
+	// BootstrapReady is the state of the bootstrap provider.
+	BootstrapReady bool `json:"bootstrapReady"`
 
-	// Infrastructure is the state of the infrastructure provider.
-	Infrastructure *string `json:"infrastructure,omitempty"`
+	// InfrastructureReady is the state of the infrastructure provider.
+	InfrastructureReady bool `json:"infrastructureReady"`
+}
+
+// SetTypedPhase sets the Phase field to the string representation of MachinePhase.
+func (m *MachineStatus) SetTypedPhase(p MachinePhase) {
```

Comment:
I can change it, thinking was that Phase is already a field and I wanted to avoid confusion for users of these methods, wdyt?

Reply:
I thought we agree that `Phase` would remain a string: https://github.com/kubernetes-sigs/cluster-api/pull/997#discussion_r292963465

Reply:
Since the commits were squashed I can't see the history. Either way, the current proposal makes the state machine explicit so I think it's fair to type it. I'll open an issue suggesting changes to the proposal out-of-band with this PR.

Reply:
Yeah, it's unchanged from v1alpha1, these methods are helpers to deal with the MachinePhase type (string)